### PR TITLE
Update Micronaut Data To latest versions

### PR DESCRIPTION
**Pull request by radovanradic**
Just to verify native tests are passing with latest micronaut-sql from snapshots

### CODE REVIEW

**Comment by radovanradic** [comment]
More issues - I think new mongodb release is needed with new micronaut-validation and micronaut-core
And r2dbc too

File: data-mongodb/src/main/java/io/micronaut/data/mongodb/conf/MongoDataConfiguration.java [hunk 1]
```diff
@@ -18,19 +18,17 @@
 import io.micronaut.context.annotation.ConfigurationProperties;
 import io.micronaut.context.condition.Condition;
 import io.micronaut.context.condition.ConditionContext;
+import io.micronaut.core.annotation.ReflectiveAccess;
 
 /**
  * Micronaut Data MongoDB integration configuration.
  */
 @ConfigurationProperties(MongoDataConfiguration.PREFIX)
```
Comment (radovanradic) [inline data-mongodb/src/main/java/io/micronaut/data/mongodb/conf/MongoDataConfiguration.java:26]: No idea why we need this now, but without it mongo native tests are failing because no bean of `MongoDataConfiguration` is found and custom conditions using it are failing.
@dstepanov

File: data-r2dbc/build.gradle [hunk 2]
```diff
@@ -21,6 +21,7 @@ dependencies {
     implementation mn.micronaut.json.core
     compileOnly mnSerde.micronaut.serde.oracle.jdbc.json
     implementation mnReactor.micronaut.reactor
+    implementation libs.managed.reactor.netty.core
```
Comment (radovanradic) [inline data-r2dbc/build.gradle:26]: Similar to https://github.com/micronaut-projects/micronaut-r2dbc/pull/655/files#r1670652324

File: data-mongodb/src/main/java/io/micronaut/data/mongodb/conf/MongoDataConfiguration.java [hunk 3]
```diff
@@ -18,19 +18,17 @@
...
+@ReflectiveAccess
 public final class MongoDataConfiguration {
 
     public static final String PREFIX = "micronaut.data.mongodb";
     public static final String CREATE_COLLECTIONS_PROPERTY = PREFIX + ".create-collections";
-    public static final String DRIVER_TYPE_PROPERTY = PREFIX + ".driver-type";
```
Comment (sdelamo) [inline data-mongodb/src/main/java/io/micronaut/data/mongodb/conf/MongoDataConfiguration.java]: @radovanradic why did you remove these?This would be a breaking change.

Comment (radovanradic) [reply data-mongodb/src/main/java/io/micronaut/data/mongodb/conf/MongoDataConfiguration.java]: I saw these were unused in the code, but forgot could be used elsewhere. Will revert

File: gradle/libs.versions.toml [hunk 4]
```diff
@@ -96,6 +98,9 @@ spring-data-commons = { module = "org.springframework.data:spring-data-commons",
 
 micronaut-azure-cosmos = { module = "io.micronaut.azure:micronaut-azure-cosmos", version.ref = "micronaut-azure" }
 
+# Reactor Netty Dependencies
```
Comment (radovanradic) [inline gradle/libs.versions.toml:98]: Should be removed when https://github.com/micronaut-projects/micronaut-r2dbc/pull/669 is done and r2dbc 5.6.1 is released

**Review by sdelamo — APPROVED** [review]
